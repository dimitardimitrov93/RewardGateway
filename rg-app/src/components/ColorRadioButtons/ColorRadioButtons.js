import * as React from 'react';
import { pink, orange, yellow } from '@mui/material/colors';
import Radio from '@mui/material/Radio';

export default function ColorRadioButtons({ setBackgroundColor }) {
	const [selectedValue, setSelectedValue] = React.useState('a');

	const handleChange = (event) => {
		setSelectedValue(event.target.value);
		setBackgroundColor(event.target.value);
	};

	const controlProps = (item) => ({
		checked: selectedValue === item,
		onChange: handleChange,
		value: item,
		name: 'color-radio-button-demo',
		inputProps: { 'aria-label': item },
	});

	return (
		<div style={{ backgroundColor: "#ffffff73", borderRadius: "1em" }}>
			<Radio
				data-testid="radio-yellow"
				{...controlProps('yellow')}
				sx={{
					color: yellow[800],
					'&.Mui-checked': {
						color: orange[600],
					},
				}}
			/>
			<Radio
				data-testid="radio-violet"
				{...controlProps('violet')}
				color="secondary"
			/>
			<Radio
				data-testid="radio-grey"
				{...controlProps('grey')}
				color="default"
			/>
			<Radio
				data-testid="radio-pink"
				{...controlProps('pink')}
				sx={{
					color: pink[800],
					'&.Mui-checked': {
						color: pink[600],
					},
				}}
			/>
			<Radio
				data-testid="radio-orange"
				{...controlProps('orange')}
				sx={{
					color: orange[800],
					'&.Mui-checked': {
						color: orange[600],
					},
				}}
			/>
		</div>
	);
}