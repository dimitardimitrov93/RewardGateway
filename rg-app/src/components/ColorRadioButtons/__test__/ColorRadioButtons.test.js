import React from 'react';
import ReactDOM from 'react-dom';
import ColorRadioButtons from '../ColorRadioButtons';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders ColorRadioButtons without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ColorRadioButtons />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders YellowRadioElement component correctly', () => {
    render(<ColorRadioButtons />);
    const YellowRadioElement = screen.getByTestId('radio-yellow');
    expect(YellowRadioElement).toBeInTheDocument();
});

it('renders VioletRadioElement component correctly', () => {
    render(<ColorRadioButtons />);
    const VioletRadioElement = screen.getByTestId('radio-violet');
    expect(VioletRadioElement).toBeInTheDocument();
});

it('renders GreyRadioElement component correctly', () => {
    render(<ColorRadioButtons />);
    const GreyRadioElement = screen.getByTestId('radio-grey');
    expect(GreyRadioElement).toBeInTheDocument();
});

it('renders OrangeRadioElement component correctly', () => {
    render(<ColorRadioButtons />);
    const OrangeRadioElement = screen.getByTestId('radio-orange');
    expect(OrangeRadioElement).toBeInTheDocument();
});

it('renders PinkRadioElement component correctly', () => {
    render(<ColorRadioButtons />);
    const PinkRadioElement = screen.getByTestId('radio-pink');
    expect(PinkRadioElement).toBeInTheDocument();
});

test('matches snapshot', () => {
    const tree = renderer.create(<ColorRadioButtons />).toJSON();
    expect(tree).toMatchSnapshot();
})

afterEach(() => {
    cleanup();
});