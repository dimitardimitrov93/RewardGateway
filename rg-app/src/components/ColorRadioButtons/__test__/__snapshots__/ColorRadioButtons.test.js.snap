// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#ffffff73",
      "borderRadius": "1em",
    }
  }
>
  <span
    className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-jlr4rk-MuiButtonBase-root-MuiRadio-root"
    data-testid="radio-yellow"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      aria-label="yellow"
      checked={false}
      className="PrivateSwitchBase-input css-1m9pwf3"
      name="color-radio-button-demo"
      onChange={[Function]}
      type="radio"
      value="yellow"
    />
    <span
      className="css-hyxlzm"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
        data-testid="RadioButtonUncheckedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
        data-testid="RadioButtonCheckedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiRadio-root MuiRadio-colorSecondary MuiButtonBase-root MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-root css-d2bh51-MuiButtonBase-root-MuiRadio-root"
    data-testid="radio-violet"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      aria-label="violet"
      checked={false}
      className="PrivateSwitchBase-input css-1m9pwf3"
      name="color-radio-button-demo"
      onChange={[Function]}
      type="radio"
      value="violet"
    />
    <span
      className="css-hyxlzm"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
        data-testid="RadioButtonUncheckedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
        data-testid="RadioButtonCheckedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiRadio-root MuiRadio-colorDefault MuiButtonBase-root MuiRadio-root MuiRadio-colorDefault PrivateSwitchBase-root css-1gu2tss-MuiButtonBase-root-MuiRadio-root"
    data-testid="radio-grey"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      aria-label="grey"
      checked={false}
      className="PrivateSwitchBase-input css-1m9pwf3"
      name="color-radio-button-demo"
      onChange={[Function]}
      type="radio"
      value="grey"
    />
    <span
      className="css-hyxlzm"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
        data-testid="RadioButtonUncheckedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
        data-testid="RadioButtonCheckedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-o22y52-MuiButtonBase-root-MuiRadio-root"
    data-testid="radio-pink"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      aria-label="pink"
      checked={false}
      className="PrivateSwitchBase-input css-1m9pwf3"
      name="color-radio-button-demo"
      onChange={[Function]}
      type="radio"
      value="pink"
    />
    <span
      className="css-hyxlzm"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
        data-testid="RadioButtonUncheckedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
        data-testid="RadioButtonCheckedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-1nkaav0-MuiButtonBase-root-MuiRadio-root"
    data-testid="radio-orange"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      aria-label="orange"
      checked={false}
      className="PrivateSwitchBase-input css-1m9pwf3"
      name="color-radio-button-demo"
      onChange={[Function]}
      type="radio"
      value="orange"
    />
    <span
      className="css-hyxlzm"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
        data-testid="RadioButtonUncheckedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
        data-testid="RadioButtonCheckedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
        />
      </svg>
    </span>
  </span>
</div>
`;
